### ISSUE TITLE
Better number handling & date handling

### ISSUE THREAD

**Issue by dathbe** [Tuxdiver/MMM-Rest#12]
I've made a two other, sort of vaguely related changes in another branch:

1) I have improved the handling of numerical formats so that the module is more likely to return a result.  Specifically, if the user specifies format %d or some variation of %f (e.g., %.1f), the module will try to find a number in whatever value is grabbed from the REST API.  For example, if "25 W" is returned, the updated module would find "25" using the parseFloat method, and return that as the result.  It also allows using the parsed number to determine ranges.  

This is particularly useful because OpenHAB now encourages users to specify units when using a number, so most of my "numbers" are actually strings containing digits and a string unit.  

If the user prefers to stick with %s (the only viable option available with the current module), there will be no change in behavior.

2) I have added a new option to specify that a returned value is expected to be a date (specifically, ISO 8601, but it may work with other date formats), and allows a user to customize how they want the date displayed using the Date().toLocaleString() method.  It works in parallel to the "range" and "string" options already in the module.  If the user, instead, specifies "dateOptions", the updated module will try to use the user-specified dateOptions to create a date string.  

As with (1), sticking with %s will result in no changed behavior.

These seem really different, but they are both about handling different data types in a smarter manner.  I was working on them both at the same time, so I only have a single branch with both changes.  But if you like one but not the other, I can back one change out.

If you like either or both, let me know and I'll create another PR.

**Comment by dathbe** [comment]
See #13

**Comment by Tuxdiver** [comment]
Thanks a lot, changes are merged!
